### PR TITLE
Sg 6 & 7 - Hex coords, map storage

### PR DESCRIPTION
**Pull request by IDontEvenCarem**
This is still only a partial implementation, although it mostly lacks future proofing and more advanced features. There's still some work do do with that, like areas of hexes, rings and circles, but I don't want to block others while developing things, that are not needed right now.

### CODE REVIEW

File: include/hex.hpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        return data.at(compute_index(wrapped));
+    }
+
+    HexT& at_ref_normalized(const HexCoords hc) {
+        return data.at(compute_index(normalized_coords(hc)));
+    }
+
+    std::vector<HexCoords> all_within_unscaled_quad 
+    (Vector2 top_left, Vector2 top_right, Vector2 bottom_left, Vector2 bottom_right) 
+    {
```
Comment (xopyip) [inline include/hex.hpp:133]: formatowanie do poprawy, bo jest bardzo niejednolite, juź lepiej by było
    std::vector<HexCoords> all_within_unscaled_quad (
Vector2 top_left,
 Vector2 top_right,
 Vector2 bottom_left,
 Vector2 bottom_right
){

Jak klamerki otwieramy w tej samej linii to nie zaczynajmy tutaj inaczej

File: src/main.cpp [hunk 2]
```diff
@@ -1,53 +1,109 @@
...
+        }
+
+        // this is temporary and also terrible, and also shows the bad frustom in all_within_unscaled_quad
+        const auto scroll = GetMouseWheelMove();
+        if (abs(scroll) > 0.01f) {
+            camera.fovy = Clamp(camera.fovy + scroll * 3.0f, 30.0f, 110.0f);
+        };
 
         BeginDrawing();
             ClearBackground(WHITE);
```
Comment (xopyip) [inline src/main.cpp:89]: Co myślisz o użyciu klamerek do wydzielenia bloków?
{
  ClearBackground(WHITE);
  ...
}
Pozwoli to na autoformatowanie kodu w poprawny sposób

File: include/hex.hpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+To get a cell, you can take an existing one, and add numbers suffixed by those UDLs
+So, for example:
+    HexCoords two_to_the_right = current + 2_R;
+*/
+
+struct HexCoords;
+struct EdgeCoords;
+
+// Operators to get offsets from a cell. They are Left and Right combined with Up, Down, or just
+HexCoords operator "" _LU (size_t x);
```
Comment (xopyip) [inline include/hex.hpp:24]: na systemie macos size_t odmawia tu posłuszeństwa z błędem:
`error: no matching literal operator for call to 'operator""_R' with argument of type 'unsigned long long' or 'const char *', and no matching literal operator template`
Czy unsigned long long zadziała crossplatformowo?

**Review by Karfrys — APPROVED** [review]
Code review ok

**Pull request merged**